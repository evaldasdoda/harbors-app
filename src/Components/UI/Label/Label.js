import React from 'react';
import PropTypes from 'prop-types';

// Styles
// require('./Input.scss');

const input = props => {
    return (
        <label htmlFor={props.for}>{props.children}</label>
    );
};

input.propTypes = {
    for: PropTypes.string
};

export default input;
