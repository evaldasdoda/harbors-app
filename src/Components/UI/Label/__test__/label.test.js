import React from 'react';
import ReactDOM from 'react-dom';
import Label from '../Label';

import {render, cleanup} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<Label></Label>, div)
})

it("renders label correctly",()=>{
    const {getByTestId} = render(<Label htmlFor="testid" > Test label </Label>)
    expect(getByTestId('label')).toHaveTextContent('Test label')
})


it("matches snapshot",()=>{
    const tree = renderer.create(
        <Label 
            htmlFor="testid"
        > 
        Test label
        </Label>).toJSON();
    expect(tree).toMatchSnapshot();
})