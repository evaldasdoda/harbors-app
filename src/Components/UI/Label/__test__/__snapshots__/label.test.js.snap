// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<label
  data-testid="label"
>
  Test label
</label>
`;
